docs/DeveloperGuide.md

```diff
@@ -394,6 +394,57 @@ The following activity diagram summarizes what happens when a user executes a ne
          - **Potential Errors**: If the list of persons changes (e.g., due to deletions or additions), the numbering index could become outdated, leading to errors.
          - **Limited Identifiability**: Index numbers do not provide any context about the person, which may be confusing when there are multiple people with the same name or similar information.
 
+### Find Feature
+
+#### Description
+
+The `FindCommand` allows users to find existing person(s) within the patient list, using their name or NRIC, and view their field data.
+
+#### Implementation Details
+
+The `FindCommand` is implemented as follows:
+- **Command Word**: The command word for this feature is `find`.
+- **Usage**: Users invoke the `FindCommand` by specifying the command word, followed by the name or NRIC of the person(s) they wish to find.
+    - The command format is: `find n/NAME` or `find id/IC_NUMBER`.
+- **`execute` method**: The `FindCommand` executes the search by using the specified predicates (`NameContainsKeywordsPredicate` or `IdContainsKeywordsPredicate`) to filter and list all persons matching the search criteria.
+- **Validation**: The `FindCommand` performs validation to ensure at least one keyword is provided. It searches based on either name or NRIC, to speed up the search and prevent possible conflicts if name and NRIC do not match each other.
+- **Execution**: When executed, the `FindCommand` identifies the person(s) being searched for based on the provided name or NRIC. If a name is provided as keyword, a `FindCommand(NameContainsKeywordsPredicate)` is created, and if an NRIC is provided as keyword, a `FindCommand(IdContainsKeywordsPredicate)` is created. `updateFilteredPersonList` will then update the filter of the filtered person list to filter by the given name or NRIC predicate (keyword).
+
+The following sequence diagram shows how the find operation works:
+
+<puml src="diagrams/FindSequenceDiagram.puml" width="400" />
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+<puml src="diagrams/FindActivityDiagram.puml" width="500" />
+
+#### Rationale
+
+- **Flexibility**: The `FindCommand` provides flexibility to users by allowing them to choose whether to edit a person by name or NRIC, whichever is faster or available.
```

Comment:
FindCommand should allow users to choose whether to find a person by name or NRIC, not edit a person.

Reply:
Thank you! I've fixed it